### PR TITLE
terminal: security fixes

### PR DESCRIPTION
**Pull request by brauner**
Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>

### CODE REVIEW

File: src/lxc/terminal.c [hunk 1]
```diff
@@ -584,6 +584,18 @@ static int lxc_terminal_peer_proxy_alloc(struct lxc_terminal *terminal,
 		goto on_error;
 	}
 
+	ret = fcntl(terminal->proxy.master, F_SETFD, FD_CLOEXEC);
```
Comment (Blub) [inline src/lxc/terminal.c]: I always wonder - given the fds come from `openpty` where you don't pass flags and don't exactly know which flags they have, if this shouldn't be a read-modify-write as stated in the `fcntl(ep)` manpage's `Application Usage` section.

Comment (brauner) [reply src/lxc/terminal.c]: You mean retrieve the flags, set the `FD_CLOEXEC` flag and pass the flags back down?

Comment (brauner) [reply src/lxc/terminal.c]: Care to push a patch on top of my branch?

Comment (brauner) [reply src/lxc/terminal.c]: Hm, or send me one :D

**Comment by brauner** [comment]
@Blub, added `fd_cloexec()`.

**Pull request merged**